File: SundeeFundee/Domain/History/HistoryItem.swift
import Foundation

// MARK: - HistoryItemSource

enum HistoryItemSource: Hashable, Sendable {
    case aiWorkout
    case program(name: String)
}

// MARK: - HistoryItem

struct HistoryItem: Identifiable, Sendable {
    let id: String
    let title: String
    let source: HistoryItemSource
    let completedAt: Date
    let exerciseCount: Int
    let durationSeconds: Int

    /// Original ID for deletion (not prefixed).
    let originalID: String
    let isAIWorkout: Bool

    var sourceLabel: String {
        switch source {
        case .aiWorkout: return "AI Workout"
        case .program(let name): return name
        }
    }
}

File: SundeeFundeeRN/src/domain/history/history-item.ts
/**
 * HistoryItem
 * Ported from SundeeFundee/Domain/History/HistoryItem.swift
 *
 * Unified type for displaying AI-generated and program-based workout records
 * in a single chronological history list.
 * Pure types + pure helper functions — no storage access.
 */

// ---------------------------------------------------------------------------
// HistoryItemSource — discriminated union matching Swift enum
// ---------------------------------------------------------------------------

export type HistoryItemSource =
  | { kind: 'aiWorkout' }
  | { kind: 'program'; name: string };

// ---------------------------------------------------------------------------
// HistoryItem
// ---------------------------------------------------------------------------

export interface HistoryItem {
  id: string;
  title: string;
  source: HistoryItemSource;
  completedAt: Date;
  exerciseCount: number;
  durationSeconds: number;
  /** Original ID for deletion (not prefixed). */
  originalID: string;
  isAIWorkout: boolean;
}

// ---------------------------------------------------------------------------
// Pure helper functions
// ---------------------------------------------------------------------------

/**
 * Returns a human-readable source label.
 * Matches Swift HistoryItem.sourceLabel computed property.
 */
export function getSourceLabel(source: HistoryItemSource): string {
  switch (source.kind) {
    case 'aiWorkout':
      return 'AI Workout';
    case 'program':
      return source.name;
  }
}
